Logging of source IP address in Sidecar routing requests

CarltonSemple opened the pull request:
I added the source IP address for each request to the logs. This updates repeatedly so that if the sidecar's IP address changes, it should show the current IP address.  
The information will be make the logs more useful since you will now be able to see the information for each connection from Source A to Destination B along with a firm way to identify which container is generating Source A

![image](https://cloud.githubusercontent.com/assets/6462396/19773951/713c24ca-9c39-11e6-9385-6bd50c3f1497.png)

CarltonSemple committed:
Added source IP address to the sidecar log for each routing request

CarltonSemple committed:
updated ubuntu sidecar dockerfile with correct steps for installing ifconfig

rshriram commented on sidecar/nginx/lua/amalgam8.lua:
```diff
@@ -476,6 +478,7 @@ function Amalgam8:new()
 
    self.myname = myname
    self.mytags = mytags
+   self.myip = "0.0.0.0"
```
Why can't you fetch the IP here?
The IP address of the container will not change once launched. The init function is executed when nginx is (re)started in the container.

CarltonSemple replied:
oh, i see.

rshriram commented on docker/Dockerfile.sidecar.ubuntu:
```diff
@@ -32,6 +32,8 @@ RUN set -ex \
     && rm -rf /tmp/filebeat*
 
+RUN apt-get update; apt-get upgrade -y
+RUN apt-get install -y net-tools
```
You don't need to install anything. IP address can be obtained in a simpler way, as shown below.

CarltonSemple replied:
alrighty, will do

rshriram commented on sidecar/nginx/lua/amalgam8.lua:
```diff
@@ -498,6 +501,8 @@ function Amalgam8:update_state(input)
    local a8_actions = {}
    local err
 
+   self.myip = self:os_cmd("ifconfig eth0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}'", false)
```
Move to init block? Doing this in update_state simply adds overhead.

Secondly, to get the local IP

``` bash
hostname -i
```

CarltonSemple replied:
i had moved it to this position for the possibility that the IP address might change. i can move it back though

CarltonSemple replied:
i tried `hostname -i` on my VM. it gave me two responses, but in one of the containers it gave me one IP

rshriram commented on sidecar/nginx/lua/amalgam8.lua:
```diff
@@ -852,5 +857,24 @@ function Amalgam8:get_mytags()
+   return self.myip
+end
+
+function Amalgam8:get_env(name)
```
This function seems unnecessary. Not used anywhere.

rshriram commented on sidecar/nginx/lua/amalgam8.lua:
```diff
@@ -852,5 +857,24 @@ function Amalgam8:get_mytags()
+   return os.getenv(name)
+end
+
+function Amalgam8:os_cmd(cmd, raw)
```
Given that you are using this only to parse the IP address, you might as well rename this function to something like ip_cmd() and you just have to return the result of the "hostname -i" without the new lines.

CarltonSemple committed:
simplified obtaining of IP address in sidecar

CarltonSemple committed:
Merge branch 'master' into side-car-log-source-ip

CarltonSemple committed:
applied filter for hostname -i

in some cases, hostname -i returns multiple IP addresses, but they are duplicates. So i'm printing just the first one

CarltonSemple commented:
i checked `hostname -i` with a container i have on bluemix, and it was returning two of the same IP address, so that's why I added a filter to the command
